File: src/G2_py/PYA203.py
year=eval(input())

# 判斷規則：每四年一閏，每百年不閏，但每四百年也一閏
if (year%4 == 0) & (year%100 != 0) | (year%400 == 0):
    print("{:} is a leap year.".format(year))
else: 
    print("{:} is not a leap year.".format(year))

File: src/G2_rust/PYA203.rs
use std::io;

fn main(){
    let mut input: String= String::new();
    io::stdin().read_line(&mut input).unwrap();
    let year: i32= input.trim().parse().unwrap();
    let result_false: i32= 0;

    if year%4 == result_false && year%100 != result_false || year%400 == result_false {
        println!("{} is a leap year.",year);
    }else{
        println!("{} is not a leap year.",year);
    }
}